samples/discoverNatBehavior.c

```diff
@@ -2,43 +2,50 @@
 
 #define NETWORK_INTERFACE_NAME_PARAM "-i"
 #define STUN_HOSTNAME_PARAM          "-s"
-#define DEFAULT_STUN_HOST            "stun:stun.sipgate.net:3478"
 
 BOOL filterFunc(UINT64 data, PCHAR name)
 {
     CHAR* desiredInterface = (CHAR*) data;
-    if (desiredInterface == NULL || STRCMP(name, desiredInterface) == 0) {
+    if (desiredInterface == NULL || STRNCMP(name, desiredInterface, STRLEN(name)) == 0) {
         return TRUE;
     }
     return FALSE;
 }
 
 INT32 main(INT32 argc, CHAR** argv)
 {
-    PCHAR interfaceName = NULL, stunHostname = NULL;
-    printf("Usage: ./discoverNatBehavior -i network-interface-name -s stun-hostname\n");
+    PCHAR interfaceName = NULL, stunHostname = NULL, pRegion = NULL, pHostnamePostfix = NULL;
     NAT_BEHAVIOR mappingBehavior = NAT_BEHAVIOR_NONE, filteringBehavior = NAT_BEHAVIOR_NONE;
     INT32 i;
     UINT32 logLevel = LOG_LEVEL_DEBUG;
     STATUS status = STATUS_SUCCESS;
     CHAR* logLevelStr = NULL;
 
+    printf("Usage: ./discoverNatBehavior -i network-interface-name -s stun-hostname\n");
+
+    if ((pRegion = GETENV(DEFAULT_REGION_ENV_VAR)) == NULL) {
+        pRegion = DEFAULT_AWS_REGION;
+    }
+    pHostnamePostfix = KINESIS_VIDEO_STUN_URL_POSTFIX;
+    // If region is in CN, add CN region uri postfix
+    if (STRSTR(pRegion, "cn-")) {
+        pHostnamePostfix = KINESIS_VIDEO_STUN_URL_POSTFIX_CN;
+    }
+    stunHostname = (PCHAR) MEMALLOC((MAX_ICE_CONFIG_URI_LEN + 1) * SIZEOF(CHAR));
+    SNPRINTF(stunHostname, MAX_ICE_CONFIG_URI_LEN + 1, KINESIS_VIDEO_STUN_URL, pRegion, pHostnamePostfix);
+
     for (i = 1; i < argc; ++i) {
-        if (STRCMP(argv[i], NETWORK_INTERFACE_NAME_PARAM) == 0) {
+        if (STRNCMP(argv[i], NETWORK_INTERFACE_NAME_PARAM, STRLEN(argv[i])) == 0) {
             interfaceName = argv[++i];
             i++;
-        } else if (STRCMP(argv[i], STUN_HOSTNAME_PARAM) == 0) {
+        } else if (STRNCMP(argv[i], STUN_HOSTNAME_PARAM, STRLEN(argv[i])) == 0) {
             stunHostname = argv[++i];
             i++;
         } else {
             printf("Unknown param %s", argv[i]);
         }
     }
 
-    if (stunHostname == NULL) {
-        stunHostname = DEFAULT_STUN_HOST;
-    }
-
     printf("Using stun host: %s, local network interface %s.\n", stunHostname, interfaceName);
```

Comment:
Nit. Use DLOGx instead of printf in this sample.

Reply:
DLOGX is not working in this application actually

Reply:
Its because the log level is being set after this :)
Move it to the top and it will work.

Reply:
Done, sorry I didn't notice that earlier